Peer detection by integration of the app-to-app communicator

boninggong opened the pull request:
Merged https://github.com/Tribler/app-to-app-communicator together with the TrustChain implementation on Android. The app is now able to find peers using the app-to-app communicator and send blocks to each other using this network. This currently only works on Eduroam/manually entering the local IP and **not** on other (public) networks. The app will be polished in a subsequent PR.

klikooo commented:
I would like to add some things:
- Could you explain what the color of each circle means and what the blinking light is?
- Could you add some documentation on how it works plus what some values are (for example in the peer list I see a carrier and next to that four digits, which I think is the ID of the carrier?)
- I see you use System.out for logging, I think it is better to use the standard logger of Android

wkmeijer commented:
So I had one moment where I could create a new block with another peer, so it does not seem to always work. The gossiping does seem to be more reliable.

Please note that since the TrustChainActivity is not the first activity that is started the app won't create a keypair right away. You can check this by clearing the database and checking the chainexplorer from the first screen, it will be empty.

boninggong commented:
First of all thank you for your comments.

-	The merge uses the app-to-app communicator to detect the public IPs of other peers. The IPs are then used in the TrustChain Android app to send blocks to each other. The app-to-app peer class (PeerAppToApp) is basically casted to a TrustChain Android peer (Peer). So they are able to mutually exchange information. The different activities of both apps are merged and slightly adjusted to make it more user friendly as well.

-	The app-to-app communicator works on other (public) networks. Peers can detect each other and make a connection through a (for now fixed) host. Unfortunately we broke the functionality of sending blocks to each other when on the same network. We will try to fix this asap. 

-	For the blinking lights, we will add a feature in the next PR with an information activity that will explain all the different colors and animations. We can also add this explanation to the documentation if preferred.

-	The four digits next to the carrier is the peer ID indeed. Currently we are adding a functionality in which a user can fill in a username and where the list of public peers will show these usernames. In the next PR we will add documentation on how this exactly works and what the values stand for.

-	The System.out logging will be removed right now. It was used in the app-to-app communicator application apparently.

-	Thank you for letting us know about the creation of the keypair. We just fixed it and will merge it to this PR.

Michieldoesburg commented:
"I can't seem to get it working by manually entering a local IP, so unfortunately I won't be able to test it properly tonight. Tomorrow I'll try it again on eduroam."

We accidentally broke connecting through local IP, but it has been fixed in commit the latest commit (
5965106).

klikooo commented:
Great! 

Is this finished now so we can check if it works properly and if we got more remarks? If not, could you notify us when it is ready?

Michieldoesburg commented:
Yes please review and try it out now this another group is waiting for this code to be merged as well.

synctext commented:
Would be great if this is acceptable for merging. Tried our the .apk release version, works for me when on eduroam (assuming 3G+4G fix in later PR).
![screenshot_20171212-144607](https://user-images.githubusercontent.com/325224/33896945-77dccf72-df64-11e7-9d9c-fdda012216ac.png)
